src/mono/mono/metadata/metadata.c

```diff
@@ -1057,7 +1057,15 @@ mono_metadata_string_heap (MonoImage *meta, guint32 index)
 const char *
 mono_metadata_string_heap_checked (MonoImage *meta, guint32 index, MonoError *error)
 {
-	if (G_UNLIKELY (!(index < meta->heap_strings.size))) {
+	if (mono_image_is_dynamic (meta))
```

Comment:
I am a little bit nervous about this.  Would feel better if this was a `g_assert (!image_is_dynamic (image));` and there was a separate function for getting strings from dynamic images.
I could be convinced otherwise, though.

@vargaz what do you think?

Reply:
I see it the other way - I accidentally used these functions expecting them to work on `MonoDynamicImage`s. But I don't have a strong preference.